lib/src/repo.rs

```diff
@@ -1386,7 +1386,7 @@ impl MutableRepo {
         self.maybe_abandon_wc_commit(&workspace_id)?;
         self.add_head(commit)?;
         self.set_wc_commit(workspace_id, commit.id().clone())
-            .map_err(|RewriteRootCommit| EditCommitError::RewriteRootCommit)
+            .map_err(|_: RewriteRootCommit| EditCommitError::RewriteRootCommit)
```

Comment:
I think the intent here is to ensure that `RewriteRootCommit` is a unit struct. It could be expressed as `|RewriteRootCommit: RewriteRootCommit|`, but that seems too verbose.

Reply:
I was also wondering if that might have been the intent. Seems unlikely that `RewriteRootCommit` will gain other variants or additional data, though.

I'm also curious to see how the merge queue deals with this unresolved comment :)

Reply:
Still it doesn't make much sense to relax the requirement. If `RewriteRootCommit` had a value, it would have to be propagated.

Reply:
How about `RewriteRootCommit: _` to make it a pattern-match and still rely on the compiler to infer the type?

Reply:
> How about RewriteRootCommit: _ to make it a pattern-match and still rely on the compiler to infer the type?

This doesn't work because as soon as you add a member to the struct the compiler starts treating `RewriteRootCommit` as a variable name.